### DISCUSSION TITLE
string_cmp_value missing UTF8 code

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#231] [Q&A]
As I look at non-strict comparisons, I found possibly an inconsistency in the string_cmp_value function:
```
static RX_INLINE int string_cmp_value(value *v1, value *v2) {
    return string_cmp(v1->string_value, v1->string_length,
                      v2->string_value, v2->string_length);
```
Shouldn't we distinguish between ASCII and UTF8?

My proposal would be:
```
#ifndef NUTF8
  static RX_INLINE int string_cmp_value(value *v1, value *v2) {
    return string_cmp(v1->string_value, v1->string_chars,
                      v2->string_value, v2->string_chars);
#else
   static RX_INLINE int string_cmp_value(value *v1, value *v2) {
    return string_cmp(v1->string_value, v1->string_length,
                      v2->string_value, v2->string_length);
#endif
```
looking further down in rxvmvars.h I found some also string_cmp_const. Is there a misunderstanding on my side?

**Comment by rvjansen** [comment]
This was the question that gave me the impression that there was work done on the non-strict comparisons. Is it resolved?

**Comment by Peter-Jacob** [comment]
you are right, we need the distinction. I am still baffled, that I should have done something in this area. I can't remember! If true, I'd doubt my state of mind.

**Reply by Peter-Jacob** [reply]
Code changed and committed:  6d015c94fcab85a03d12f3e3934b4b8b756d558a. The results are now:
```
random(1,999,8142) 653
random(1,999,8142) 653
random(1,999) 715
random(1,999) 185
random(1,999) 72
random(1,999) 742
random(1,999,8142) 653
random(1,999) 715
random(1,999) 185
random(1,999) 72
random(1,999) 742
```

**Reply by rvjansen** [reply]
looking great!
